src/edu/csus/ecs/pc2/core/InternalController.java

```diff
@@ -4988,22 +4988,50 @@ public IInternalContest getContest() {
     }
 
     @Override
-    public void submitRun(ClientId submitter, Problem problem, Language language, SerializedFile mainSubmissionFile, SerializedFile[] additionalFiles, long overrideTimeMS, long overrideRunId) {
+    public void submitRun(ClientId submitter, Problem problem, Language language, SerializedFile mainSubmissionFile, SerializedFile[] additionalFiles, long overrideTimeMS, long overrideRunId) throws Exception {
 
         submitRun(submitter, problem, language, null, mainSubmissionFile, additionalFiles, overrideTimeMS, overrideRunId);
     }
 
     @Override
-    public void submitRun(ClientId submitter, Problem problem, Language language, String entry_point, SerializedFile mainSubmissionFile, SerializedFile[] additionalFiles, long overrideTimeMS, long overrideSubmissionId) {
+    public void submitRun(ClientId submitter, Problem problem, Language language, String entry_point, SerializedFile mainSubmissionFile, SerializedFile[] additionalFiles, long overrideTimeMS, long overrideSubmissionId) throws Exception {
```

Comment:
Ditto; should throw `SubmissionRejectedException`.

Reply:
I just did what eclipse suggested.  I guess if an other exception occurs the submissions was not accepted anyway.